File: src/Lessons/2-Arrays/CyclicRotation.go
package main

import "fmt"

func main() {
	fmt.Println(solution([]int{1, 2, 3, 4, 5}, 0))
}

func solution(someArray []int, rotationTimes int) []int {
	//Rotate the array the given times
	//if rotation is zero or the same as the len of array no change will appear
	if rotationTimes == 0 || len(someArray) == rotationTimes {
		return someArray
	}
	//To avoid unnecessary long loop
	if rotationTimes > len(someArray) {
		//we'll rotate the remainder number of times
		rotationTimes = rotationTimes % len(someArray)
	}
	for index := 0; index < rotationTimes; index++ {
		//remove the last item and append it as the first
		rotateInt := []int{someArray[len(someArray)-1]} //convert to slice
		someArray = append(rotateInt, someArray[:len(someArray)-1]...)
	}
	return someArray
}

File: src/Lessons/2-Arrays/CyclicRotation.js
//Cyclic Rotation
function solution(A,K){
    //An array to be shifted right by K times
    let rotationTimes = K;
    let integerArray = A;
    //if k is zero or less return the original array no change
    if(rotationTimes <= 0 || rotationTimes == integerArray.length)
        return integerArray;

    if(rotationTimes > integerArray.length) {
        //Avoid unneccessary long loop
        rotationTimes = rotationTimes % integerArray.length;
    }
    for (let index = 0; index < rotationTimes; index++) {
        const temp = integerArray.pop()  
        integerArray.unshift(temp);
    }

    return integerArray;
}

console.log(solution([3, 8, 9, 7, 6],1))